File: stream-chat-android-docs/src/main/java/io/getstream/chat/docs/java/helpers/MyFileUploader.java
package io.getstream.chat.docs.java.helpers;


import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.io.File;

import io.getstream.chat.android.client.uploader.FileUploader;
import io.getstream.chat.android.client.utils.ProgressCallback;
import io.getstream.chat.android.client.utils.Result;
import kotlin.Unit;

public class MyFileUploader implements FileUploader {
    @Nullable
    @Override
    public Result<String> sendFile(@NotNull String channelType, @NotNull String channelId, @NotNull String userId, @NotNull String connectionId, @NotNull File file, @NotNull ProgressCallback callback) {
         try {
            return Result.success("url");
        } catch (Exception e) {
            return Result.error(e);
        }
    }

    @Nullable
    @Override
    public Result<String> sendFile(@NotNull String channelType, @NotNull String channelId, @NotNull String userId, @NotNull String connectionId, @NotNull File file) {
        try {
            return Result.success("url");
        } catch (Exception e) {
            return Result.error(e);
        }
    }

    @Nullable
    @Override
    public Result<String> sendImage(@NotNull String channelType, @NotNull String channelId, @NotNull String userId, @NotNull String connectionId, @NotNull File file, @NotNull ProgressCallback callback) {
        try {
            return Result.success("url");
        } catch (Exception e) {
            return Result.error(e);
        }
    }

    @Nullable
    @Override
    public Result<String> sendImage(@NotNull String channelType, @NotNull String channelId, @NotNull String userId, @NotNull String connectionId, @NotNull File file) {
        try {
            return Result.success("url");
        } catch (Exception e) {
            return Result.error(e);
        }
    }

    @Override
    public Result<Unit> deleteFile(@NotNull String channelType, @NotNull String channelId, @NotNull String userId, @NotNull String connectionId, @NotNull String url) {
        try {
            return Result.success(Unit.INSTANCE);
        } catch (Exception e) {
            return Result.error(e);
        }
    }

    @Override
    public Result<Unit> deleteImage(@NotNull String channelType, @NotNull String channelId, @NotNull String userId, @NotNull String connectionId, @NotNull String url) {
        try {
            return Result.success(Unit.INSTANCE);
        } catch (Exception e) {
            return Result.error(e);
        }
    }
}

File: stream-chat-android-docs/src/main/kotlin/io/getstream/chat/docs/kotlin/helpers/MyFileUploader.kt
package io.getstream.chat.docs.kotlin.helpers

import io.getstream.chat.android.client.uploader.FileUploader
import io.getstream.chat.android.client.utils.ProgressCallback
import io.getstream.chat.android.client.utils.Result
import java.io.File

class MyFileUploader : FileUploader {
    override fun sendFile(
        channelType: String,
        channelId: String,
        userId: String,
        connectionId: String,
        file: File,
        callback: ProgressCallback,
    ): Result<String> {
        return try {
            Result.success("url")
        } catch (e: Exception) {
            Result.error(e)
        }
    }

    override fun sendFile(
        channelType: String,
        channelId: String,
        userId: String,
        connectionId: String,
        file: File,
    ): Result<String> {
        return try {
            Result.success("url")
        } catch (e: Exception) {
            Result.error(e)
        }
    }

    override fun sendImage(
        channelType: String,
        channelId: String,
        userId: String,
        connectionId: String,
        file: File,
        callback: ProgressCallback,
    ): Result<String> {
        return try {
            Result.success("url")
        } catch (e: Exception) {
            Result.error(e)
        }
    }

    override fun sendImage(
        channelType: String,
        channelId: String,
        userId: String,
        connectionId: String,
        file: File,
    ): Result<String> {
        return try {
            Result.success("url")
        } catch (e: Exception) {
            Result.error(e)
        }
    }

    override fun deleteFile(
        channelType: String,
        channelId: String,
        userId: String,
        connectionId: String,
        url: String,
    ): Result<Unit> {
        return try {
            Result.success(Unit)
        } catch (e: Exception) {
            Result.error(e)
        }
    }

    override fun deleteImage(
        channelType: String,
        channelId: String,
        userId: String,
        connectionId: String,
        url: String,
    ): Result<Unit> {
        return try {
            Result.success(Unit)
        } catch (e: Exception) {
            Result.error(e)
        }
    }
}
